docs/docsite/rst/reference_appendices/faq.rst

```diff
@@ -377,7 +377,7 @@ via a role parameter or other input.  Variable names can be built by adding stri
 
 .. code-block:: jinja
 
-    {{ hostvars[inventory_hostname]['ansible_' + which_interface]['ipv4']['address'] }}
+    {{ hostvars[inventory_hostname]['ansible_' + which_interface | replace('_', '-') ]['ipv4']['address'] }}
```

Comment:
How about keeping this example as-is, but adding a `.. note::` at the end of the FAQ entry mentioning that for interface names with dashes in the example above, `| replace(...)` is needed? That would give the correct answer without making the example more complex than it should be.